plugins/PrivacyManager/Config.php

```diff
@@ -62,30 +74,48 @@ public function __get($name)
         return $this->getFromTrackerCache($name, $this->properties[$name]);
     }
 
-    private function prefix($optionName)
+    private function prefix(string $optionName, bool $addIdSite = true)
     {
-        return 'PrivacyManager.' . $optionName;
+        // if requested, adding the site ID in the middle to have all the site-specific settings together
+        return 'PrivacyManager.' . (($addIdSite && $this->idSite) ? "idSite($this->idSite)." : '') . $optionName;
     }
 
-    private function getFromTrackerCache($name, $config)
+    private function getFromSpecificTrackerCache(string $name, array $cache, array $config, bool $useFallback = true)
     {
-        $name  = $this->prefix($name);
-        $cache = Cache::getCacheGeneral();
-
         if (array_key_exists($name, $cache)) {
             $value = $cache[$name];
             settype($value, $config['type']);
 
             return $value;
         }
 
-        return $config['default'];
+        return $useFallback ? $config['default'] : null;
     }
 
-    private function getFromOption($name, $config)
+    private function getFromTrackerCache($name, $config)
     {
-        $name  = $this->prefix($name);
-        $value = Option::get($name);
+        $generalCache = Cache::getCacheGeneral();
+        $name = $this->prefix($name, false); // when getting from tracker cache, we always want the generic name
+        if ($this->idSite) {
+            $cache = Cache::getCacheWebsiteAttributes($this->idSite);
+        } else {
+            $cache = $generalCache; // so that we always have some cache to check below
+        }
+
+        // check specific cache first, if no value found there return from general cache or use default
+        $valueSite = $this->getFromSpecificTrackerCache($name, $cache, $config, $useFallback = false);
+        $valueGeneralWithFallback = $this->getFromSpecificTrackerCache($name, $generalCache, $config);
+
+        return (null !== $valueSite ? $valueSite : $valueGeneralWithFallback);
```

Comment:
Wouldn't that be the same
```suggestion
        return $valueSite ?? $valueGeneralWithFallback;
```